prow/repoowners/repoowners.go

```diff
@@ -46,6 +46,10 @@ const (
 
 type dirOptions struct {
 	NoParentOwners bool `json:"no_parent_owners,omitempty"`
+	// AllowFolderCreation will result in changes to a subpath of a given path
+	// that does not have an OWNERS file being auto-approved. This should be
+	// enabled with caution.
+	AllowFolderCreation bool `json:"allow_folder_creation,omitempty"`
```

Comment:
Can we choose a different name here? `AutoApprove`?

I don't understand why "allow folder creation" would mean "auto approve files under this folder" or "do not require files under here to need approval"

Reply:
AutoAllow, AllowFolder, whatever. I'm primarily getting caught on the "Creation" word

Reply:
Generally I am definitely open for suggestions on how to improve the name. The Creation stems from the fact that this option is not supposed to get inherited, i.E. if you have folder/OWNERS with this enabled and folder/subfolder/OWNERS, then changes to folder/subfolder should _not_ get autoapproved. 

Maybe `AutoApproveSubfoldersIfNoOwners`?

Reply:
So if I set AllowFolderCreation for folder/OWNERS then folder/foo/whatever.go gets autoapproved because there is no folder/foo/OWNERS whereas folder/bar/something.go needs approval from someone (/OWNERS, folder/OWNERS, folder/bar/OWNERS) because folder/bar/OWNERS exists?

And furthermore that folder/bar/more/this.go will need the same approval (unless folder/bar/OWNERS or folder/bar/more/OWNERS.go sets this flag?)

Reply:
Yes, exactly. The underlying usecase is to allow ppl to PR new config in without requiring approval but not to allow them to mess with other ppls config. In most cases this will be completed by additional verification that checks that the new config doesn't override existing config. It would for example help with the delegation story in https://docs.google.com/document/d/1S8DFXrvFw27xYyHMDZmtUkVvd0cbDzFI8T5H35FkCyU/edit# (currently listed as "possibly openshift-specific" but it appears that ppl think it's a good idea to do it upstream)

Reply:
Gotcha! This feature sounds great.

Reply:
I do not at all understand the link between "Creation" and "Not inherited by children." Can we drop that word?

Personally I think just documenting that a child OWNERS file (and the files owned by that child file) does not inherit this feature is sufficient (without adding Creating to the word).

Your doc comment calls this "auto-approved", how about using that word?

AKA `AutoApprove` `AutoApproveSubfolders` or `AutoApproveChildren` whatever.

Reply:
Renamed to `AutoApproveSubfolders`

Reply:
I've renamed it yet again to `AutoApproveUnownedSubfolders` to clarify that it will not apply when a subfolder has an owners file, WDYT?